docs/table_design/expression_partitioning.md

```diff
@@ -0,0 +1,256 @@
+
+expression ::=
+    ( partition_columns )
+    
+partition_columns ::=
+    <column>, [ <column> [,...] ]
+```
+
+### Parameters
+
+| **Parameters**          | **Required** | **Description**                                                  |
+| ----------------------- | -------- | ------------------------------------------------------------ |
+| `partition_columns`     | YES      | The names of partition columns.<br> <ul><li>The partition column values can be string (BINARY not supported), date or datetime, integer, and boolean values. The partition column allows `NULL` values.</li><li> Each partition can only contain data with the same value for a partition column. To include data with different values in a partition column in a partition, see [List partitioning](./list_partitioning.md).</li></ul> |
+| `partition_live_number` | YES      | The number of partitions to be retained. Compare the values of partition columns among partitions, and periodically delete partitions with smaller values while retaining those with larger values.<br>StarRocks schedules tasks to manage the number of partitions, and the scheduling interval can be configured through the FE dynamic parameter `dynamic_partition_check_interval_seconds`, which defaults to 600 seconds (10 minutes).<br>**NOTE**<br>If the values in the partition column are strings, StarRocks compare the lexicographical order of the partition names, and periodically retains the partitions that come earlier while deleting the partitions that come later. |
+
+### Usage notes
+
+- During data loading, StarRocks automatically creates some partitions based on the loaded data, but if the load job fails for some reason, the partitions that are automatically created by StarRocks cannot be automatically deleted.
+- StarRocks sets the default maximum number of automatically created partitions to 4096, which can be configured by the FE parameter `max_automatic_partition_number`. This parameter can prevent you from accidentally creating too many partitions.
+- The naming rule for partitions: if multiple partition columns are specified, the values of different partition columns are connected with an underscore `_` in the partition name, and the format is `p<value in partition column 1>_<value in partition column 2>_...`.  For example, if two columns `dt` and `province` are specified as partition columns, both of which are string types, and a data row with values `2022-04-01` and `beijing` is loaded, the corresponding partition automatically created is named `p20220401_beijing`.
+
+### Examples
+
+Example 1: Suppose you frequently details of the query data center billing based on time ranges and specific cities. At table creation, you can use partition expressions to specify the first partition columns as `dt`  and `city` . This way, data belonging to the same date and city are routed into the same partition, and partition pruning can be used to significantly improve query efficiency.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/bca71914-fd93-4721-a171-21c3a8dbead1)
the word "query" is placed in a wrong place in the sentence. "query" should be placed preceding "details".